okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -854,6 +882,7 @@ public void settings(boolean clearPrevious, Settings settings) {
         if (firstSettings) {
           listener.transportReady();
           firstSettings = false;
+          cancelDeadlineWatchdog();
```

Comment:
Also cancel the task when the transport fails? Otherwise we could accumulate useless timers.